### PR TITLE
Support kata style vm template

### PR DESCRIPTION
**Pull request by luckyh**
Let some of the qemu opitons to be configurable, in order to support kata style vm template.

Signed-off-by: Xu Han <xuhan@redhat.com>

ID: 1902565

### CODE REVIEW

**Comment by luckyh** [comment]
### isa-debugcon
* default
```
    -chardev socket,id=seabioslog_id_20201130-002724-hUqKyMRc,path=/var/tmp/avocado_cvefo151/seabios-20201130-002724-hUqKyMRc,server,nowait \
    -device isa-debugcon,chardev=seabioslog_id_20201130-002724-hUqKyMRc,iobase=0x402 \
```
* w/ `enable_debugcon = no`
```
(removed)
```

**Comment by luckyh** [comment]
### RTC
* default
```
    -rtc base=utc,clock=host,driftfix=slew  \
```
* w/ `del rtc_base
del rtc_clock
del rtc_drift`
```
(removed)
```

**Comment by luckyh** [comment]
### accel kvm
* default
```
    -machine q35 \
...
    -enable-kvm \
```
* w/ `vm_accelerator = kvm`
```
    -machine q35,accel=kvm \
```

**Comment by luckyh** [comment]
@yanan-fu @xiangchunfu would you please help review the PR? thanks!

**Comment by xiangchunfu** [comment]
Base on previous discussion. It looks good to me. 
Acked by:Xiangchun Fu xfu@redhat.com

File: virttest/shared/cfg/base.cfg [hunk 1]
```diff
@@ -12,6 +12,9 @@ qemu_io_binary = qemu-io
 #enable_kvm = yes
 # Explicitly pass -no-kvm to qemu (default no)
 #disable_kvm = no
+# For enabling an accelerator explicitly (default ''). This option has
+# higher priority than `enable_kvm` or `disable_kvm`.
+#vm_accelerator =
```
Comment (yanan-fu) [inline virttest/shared/cfg/base.cfg:17]: This is used as a parameter for `-machines`, so prefer `machine_accel`. 
There is another `accel` device:
```
       -accel name[,prop=value[,...]]
              This is used to enable an accelerator. Depending on the target architecture, kvm, xen, hax, hvf, whpx or
              tcg can be available. By default, tcg is used. If there is more than one accelerator specified, the next
              one is used if the previous one fails to initialize.
```

File: virttest/qemu_devices/qcontainer.py [hunk 2]
```diff
@@ -1418,20 +1421,26 @@ def machine_other(cmd=False):
                                                (machine_type))
         else:
             devices = None
+            machine_opts = []
```
Comment (yanan-fu) [inline virttest/qemu_devices/qcontainer.py:1424]: The `else` blocker have existing issues, but the new added machine_accel logic is ok, so keep it is ok for me.

File: virttest/shared/cfg/base.cfg (hunk 1 above)
Comment (yanan-fu) [reply virttest/shared/cfg/base.cfg:17]: As we discussed, use `vm_accelerator` for high leave meaning, while implement with machine type parameter now.

**Review by yanan-fu — APPROVED** [review]
Acked-by: Yanan Fu <yfu@redhat.com>

File: virttest/qemu_vm.py [hunk 3]
```diff
@@ -1174,14 +1174,14 @@ def add_rtc(devices):
             # Pay attention that rtc-td-hack is for early version
             # if "rtc " in help:
             if devices.has_option("rtc"):
-                cmd = " -rtc base=%s" % params.get("rtc_base", "utc")
-                cmd += _add_option("clock", params.get("rtc_clock", "host"))
-                cmd += _add_option("driftfix", params.get("rtc_drift", None))
-                return cmd
+                cmd = _add_option("base", params.get("rtc_base"))
+                cmd += _add_option("clock", params.get("rtc_clock"))
+                cmd += _add_option("driftfix", params.get("rtc_drift"))
```
Comment (pevogam) [inline virttest/qemu_vm.py:1179]: It is really nice to see many of these previously hard-coded options now configurable.

File: virttest/qemu_vm.py [hunk 4]
```diff
@@ -1884,7 +1884,8 @@ def sort_key(dev):
             add_virtio_rng(devices, rng_params, parent_bus)
 
         # Add logging
-        devices.insert(StrDev('isa-log', cmdline=add_log_seabios(devices)))
+        if params.get("enable_debugcon") == "yes":
+            devices.insert(StrDev('isa-log', cmdline=add_log_seabios(devices)))
```
Comment (pevogam) [inline virttest/qemu_vm.py:1888]: This for instance is an option we never used so far so I am glad we can now disable it if we don't need it.

**Review by pevogam — APPROVED** [review]
LGTM, really good idea for a PR @luckyh.

**Comment by luckyh** [comment]
Thanks all!

**Pull request merged**